src/App.js

```diff
@@ -1,27 +1,45 @@
 import './App.css';
 import { Container, Logo, Header, Search, ShoppingCart, Content, Footer} from './App.style';
+import { useState } from 'react';
 
 import Home from 'components/Home/Home';
+import ProductList from 'components/ProductList/ProductList.styles'
 
 import LogoImg from 'images/image.png';
 import ShoppingCartIcon from 'images/shopping-cart.svg';
 
+import productsData from 'mocks/products.json';
```

Comment:
I suppose we are moving the raw data inside hooks because we are replacing the raw data with API fetch calls later in the bootcamp, right?

Reply:
Sorry for the late response, but yes you've got the point. You are later getting the data from an API but also because what you wrote on the first edit, it reduces a lot the properties that you don't need. Also it can be helpful in some cases where you use an api that has a quota limit, I think this is not the case for this bootcamp but in previous ones the API that was used was the one from youtube and was easily passed. In those cases what people did was to create an env variable that allows them to switch from requesting the data from the api or from the mock.
You can do something like this
```
// in the component where you are getting the data
const service = DEVENV ? YoutubeService.devSearch : YoutubeService.search; // here you go and get how you want to get the data
const { data, loading } = useFetch(service, search);
```
What is on youtube service is a bit like this. And as you can see both go get some data and parse it with the same function, the only thing they do different is one gets it from the api and one from a mock. Also both of them return a promise because here i'm only declaring the functions
```
devSearch: () => {
    return new Promise((resolve, reject) => {
      setTimeout(() => {
        const data = mapVideosData(videosData.items);
        resolve(data);
      }, 500);
    });
  },
search: async (query, maxResults = 20) => {
    try {
      const REQUESTURL = `${BASEURL}search?part=snippet&q=${query}&maxResults=${maxResults}${KEY}`;
      const response = await fetch(REQUESTURL);
      const data = await response.json();
      const vids = mapVideosData(data.items);
      return vids;
    } catch (error) {
      throw new Error(error);
    }
  }
```
Then the fetch hook can be like this. You pass the function where you want to get the data and the "query" or the search term. 
```
export const useFetch = (service, query = '') => {
  const [data, setData] = useState({
    data: null,
    loading: true,
    error: null,
  });
  useEffect(() => {
    service(query)
      .then((response) => {
        setData({
          data: response,
          loading: false,
          error: null,
        });
      })
      .catch((error) => {
        setData({
          data: null,
          loading: false,
          error,
        });
      });
  }, [service, query]);

  return data;
};
```

Reply:
Sorry for the very long response. There are multiple ways to solve this type of problems and this one can be improved a lot it's just a very quick solution I did once

Reply:
No problem, thanks a lot for the detailed response!